scripts/gpu_support/nvidia/install_cuda_and_libraries.sh

```diff
@@ -112,7 +112,14 @@ for EASYSTACK_FILE in ${TOPDIR}/easystacks/eessi-*CUDA*.yml; do
     unset EESSI_USER_INSTALL
     export EESSI_SITE_INSTALL=1
     module unload EESSI-extend
-    module load EESSI-extend/${EESSI_VERSION}-easybuild
+
+    # The EESSI-extend module is being loaded (or installed if it doesn't exist yet).
+    # The script requires the EESSI_VERSION given as argument, a couple of
+    # environment variables set (TMPDIR, EB and EASYBUILD_INSTALLPATH) and the
+    # function check_exit_code defined.
+    # NOTE, the script exits if those variables/functions are undefined.
+    export EASYBUILD_INSTALLPATH=${EESSI_PREFIX}/software/${EESSI_OS_TYPE}/${EESSI_SOFTWARE_SUBDIR_OVERRIDE}
```

Comment:
This doesn't look right, this is setting up an installation path that end users definitely do not have write access to

Reply:
I agree, either it is available or we fail. In fact this should never need to run unless we have an accelerator build, which can only happen if `EESSI-extend` is already available (since that is CPU only)

Reply:
The quick thing you would to shift https://github.com/EESSI/software-layer/pull/810/files#diff-2ca195317c2e462806b593507f473145e7499208c16b56dc7fba2e47284f4a74R250 down a bit below the first call of `load_eessi_extend_module.sh` and then you can leave things as  they were.

Reply:
Reordered code as suggested in https://github.com/EESSI/software-layer/pull/810/commits/d1264e11e6dac91e3e87ba0b35a658426fa9e3c0

Reply:
This can then be reverted to the original code, no need for changes any more

Reply:
Ack. Will change it back.

Reply:
Ah, can't revert it back. Originally the procedure was like this

1. run `install_cuda_host_injections.sh` script that uses EasyBuild with a single easyconfig
2. set up build environment (load EasyBuild, `source configure_easybuild`)

Changing this to use EESSI-extend and easystack files
1. run `install_cuda_and_libraries.sh` that loads EasyBuild, loads EESSI-extend and then processes easystack files with `CUDA` in their names
2. set up build environment (load EasyBuild, load EESSI-extend)

didn't work because the `install_cuda_and_libraries.sh` script now relies on EESSI-extend and just loading it is not sufficient if it is not there yet (eg. when a new stack is started or when it is being rebuilt).

Commit https://github.com/EESSI/software-layer/commit/d1264e11e6dac91e3e87ba0b35a658426fa9e3c0 reordered this to
1. Set up build environment (load EasyBuild, load/install EESSI-extend) for the general script
2. run `install_cuda_and_libraries.sh` that loads EasyBuild, loads EESSI-extend and then processes easystack files with `CUDA` in their names

It may seem a bit odd, that we load EasyBuild and load/install EESSI-extend both in `EESSI-install-software.sh` and in `install_cuda_and_libraries.sh`. However, doing so in `EESSI-install-software.sh` serves two purposes:
1. We ensure that both EasyBuild and EESSI-extend are available for the remainder of the script
2. Set up the build environment for the remainder of the script

Then, everything is set up for running `install_cuda_and_libraries.sh`.

However, `install_cuda_and_libraries.sh` is distributed via EESSI (`/cvmfs/software.eessi.io/versions/2023.06/scripts/gpu_support/nvidia/install_cuda_and_libraries.sh`) so that admins can install the required parts of CUDA SDK and libraries under `/host_injections/`. When they use the script, they need to have access to EasyBuild and EESSI-extend. Therefore we need to load EasyBuild and load/install EESSI-extend also in that script to be sure the build environment is set up correctly.

Reply:
We might change what we do in the `install_cuda_and_libraries.sh` script, i.e., only load the EESSI-extend module ... so we don't continue if it is not available. In `EESSI-install-software.sh` we have to `source load_eessi_extend_module.sh` or it won't be available neither for the installation of CUDA nor any other software.

Reply:
Right, I agree here. The necessary changes are:
* Introduce `load_eessi_extend_module.sh` so we can ensure we always have `EESSI-extend` availabe
* Move the use of `install_cuda_and_libraries.sh` to after we know that `EESSI-extend` exists in the build script. Using `load_eessi_extend_module.sh` in `install_cuda_and_libraries.sh` is not helpful (from an end user perspective, not our perspective), as if it were ever the case that `EESSI-extend` were not available the script would fail anyway as the end user would not have write permissions to the installation path